File: ceaser cipher decryption.py
def decrypt(message, key):
    decrypted_message = ""
    for char in message:
        if char.isupper():
            decrypted_message += chr((ord(char) - key - 65) % 26 + 65)
        elif char.islower():
            decrypted_message += chr((ord(char) - key - 97) % 26 + 97)
        else:
            decrypted_message += char
    return decrypted_message
def main():
    encrypted_message = input("Enter the encrypted message: ")
    key = int(input("Enter the key (an integer): "))
    decrypted_message = decrypt(encrypted_message, key)
    print("Decrypted message:", decrypted_message)
if __name__ == "__main__":
    main()

File: ceaser cipher decryption.cpp
#include <stdio.h>
#include <string.h>
void decrypt(char encryptedMessage[], int key) {
    int i;
    char decryptedMessage[100];
    for (i = 0; i < strlen(encryptedMessage); i++) {
        if (encryptedMessage[i] >= 'A' && encryptedMessage[i] <= 'Z') {
            decryptedMessage[i] = (encryptedMessage[i] - key - 'A' + 26) % 26 + 'A';
        }
        else if (encryptedMessage[i] >= 'a' && encryptedMessage[i] <= 'z') {
            decryptedMessage[i] = (encryptedMessage[i] - key - 'a' + 26) % 26 + 'a';
        }
        else {
            decryptedMessage[i] = encryptedMessage[i];
        }
    }
    decryptedMessage[i] = '\0'; // Null-terminate the decrypted string
    printf("Decrypted message: %s\n", decryptedMessage);
}
int main() {
    char encryptedMessage[100];
    int key;
    printf("Enter an encrypted message: ");
    fgets(encryptedMessage, sizeof(encryptedMessage), stdin);
    encryptedMessage[strcspn(encryptedMessage, "\n")] = '\0';
    printf("Enter the key (an integer): ");
    scanf("%d", &key);
    decrypt(encryptedMessage, key);
    return 0;
}
